rolldown-contract/src/rolldown.sol

```diff
@@ -279,6 +284,30 @@ contract RollDown {
         }
     }
 
+     function process_pending_update_deposit(
+        uint256 requestId,
+        bool success
+    ) private {
+        if (!success) {
+            Deposit memory newDeposit = deposits[requestId];
+
+            IERC20 token = IERC20(newDeposit.tokenAddress);
+            require(
```

Comment:
it does nothing, just attach ierc20 interface to that address. so it can call fn transfer. there is a task for ensuring accepted tokens are erc20, not implemented further then 

 IERC20 token = IERC20(tokenAddress);
         // TODO: uncomment
         // require transfer tokens from the sender to the contract
        require(
            token.transferFrom(msg.sender, tokenAddress, amount),
            "Token transfer failed"
        );
        
        in deposit. so it must have transferFrom interface, else there wouldn't be a deposit to be processed

Reply:
my point is, please add unit test that ensures that you cannot withdraw more than has been deposited

Reply:
this is what i mean by `test it in ut` ;p

https://github.com/mangata-finance/eigen-layer-monorepo/blob/feature/handle-deposit-failure/rolldown-contract/test/rolldown.t.sol#L60-L63